### DISCUSSION TITLE
Changing plugins in the repo

### DISCUSSION THREAD

**Discussion by Coffee2Bits** [DevOps-MBSE/AaC#139] [Archive]
We should start discussing changing the layout of plugins in the repository, and the difference between 1st party (AaC Developers) plugins and 3rd party plugins. The current layout has several detriments like no GHA automated unit tests. If we integrate 1st party plugins into the AaC package then we can better support them and monitor them for issues; all things that greatly benefit our users.

### Moving the plugins into the repository
I propose that we move all "1st party" plugins into the AaC package python source under a `plugins` module. The current module `plugin.py` will be changed to be the `__init__.py` for a directory module called `plugins`. This would mean that the GenProtobuf plugin's new package would be `aac.plugins.aac_gen_protobuf`.

The plugin AaC DSL would be moved into the plugin directory, `aac.plugins.aac_gen_protobuf` per our example. 

The plugin readme would be moved into the `docs` directory.

The tests would be moved in the tests directory, into a directory submodule `tests.plugins` under the plugins name e.g. `tests.plugins.aac_gen_protobuf` 

The plugins are updated to return their packaged AaC DSL yaml files instead of hardcoded strings.

The plugin's `setup.py` will need to have any unique dependencies floated up to AaC and then discard the file.

All 1st party plugins will need to be registered similar to how genjson and genplugin are registered.

### Updating GenPlugin
GenPlugin will now need to provide support for 1st and 3rd party plugin generation, since 3rd party users will need a setup very similar to the current GenPlugin generation output. 

### Other Items
GenJson and GenPlug will need to be updated to the new 1st party plugin style.

**Comment by Coffee2Bits** [comment]
This discussion is moot with #143 .

**Comment by lizzcondrey** [comment]
This has been marked resolved and has the issue and merge linked. Closing this discussion.
